jumpscale/sals/process/__init__.py

```diff
@@ -84,595 +141,833 @@ def execute(
+            ex: get_user_processes func, or get_similar_processes.
+            if not specified, psutil.process_iter will be used. Defaults to None.
+        include_zombie (bool, optional): Whether to include pid for zombie proccesses or not. Defaults to False.
+        full_cmd_line (bool, optional): The pattern is normally only matched against the process name.
+            if full_cmd_line is set to True, the full command line is used. Defaults to False.
 
     Returns:
-        [list(int)] -- list of pids
+        list of int: list of PID(s)
     """
     # default match predicate
-    # why aren't we using psutil ??
     def default_predicate(target, given):
-        return target.strip().lower() in given.lower()
+        if isinstance(given, list):
+            return target in given
+        else:
+            return target.strip().lower() == given.lower()
 
-    if match_predicate is None:
-        match_predicate = default_predicate
+    default_processes_source = psutil.process_iter(["name", "exe", "cmdline"])
 
-    if process_name is None:
-        raise j.exceptions.RuntimeError("process cannot be None")
-    if j.data.platform.is_unix():
-        pids = set()
-        for process in get_processes():
-            try:
-                pid = process.pid
-                if not isinstance(pid, int):
-                    continue
-                name = process.name()
-                if match_predicate(process_name, name):
-                    pids.add(pid)
-                elif match_predicate(process_name, process.exe()):
-                    pids.add(pid)
-                else:
-                    cmdline = process.cmdline()
-                    if cmdline and cmdline[0]:
-                        if match_predicate(process_name, cmdline[0]):
-                            pids.add(pid)
-            except (psutil.Error, FileNotFoundError):
+    match_predicate = match_predicate or default_predicate
+    p_source = _alt_source or default_processes_source
+
+    pids = []
+    for proc in p_source:
+        try:
+            if not include_zombie and proc.status() == psutil.STATUS_ZOMBIE:
+                j.logger.debug(f"ignoring : {proc.pid} zombie process.")
                 continue
-        return list(pids)
-    else:
-        raise j.exceptions.NotImplemented("getProcessPid is only implemented for unix")
+            if (
+                (match_predicate(process_name, proc.info["name"]))
+                or (proc.info["exe"] and match_predicate(process_name, os.path.basename(proc.info["exe"])))
+                or (
+                    proc.info["cmdline"]
+                    and match_predicate(
+                        process_name,
+                        proc.info["cmdline"] if full_cmd_line else os.path.basename(proc.info["cmdline"][0]),
+                    )
+                )
+            ):
+                pids.append(proc.pid)
+                j.logger.debug(f"process {len(pids) }found: {proc.info}")
```

Comment:
IMHO, these inner debug logs like this can be removed after making sure everything works.